### DISCUSSION TITLE
[Chapter 08] n차 캐시?

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#121] [Chapter 08. Data Persistence] [answered]
> JPA/Hibernate 프레임워크의 영속성 컨텍스트는 1차 캐시 역할을 한다.
> (...)
> 영속성 컨텍스트는 멀티 스레드가 공유할 수 있는 공유 자원이 아니다. 그래서 스레드마다 별도의 영속성 컨텍스트가 생성된다.
>
> 8.8.2 영속성 컨텍스트의 특징 중에서

영속성 컨텍스트(PersistenceContext)는 보통 1차 캐시라고 흔히 말하는데, 그럼 2차, 3차 같은 캐시도 있는 건가?
만약 1차 캐시 다음 단계의 캐시가 있다면 어떤 종류와 상황에서 사용해야 하는지 궁금합니다.

**Comment by Irisation23** [comment, marked as answer]
https://medium.com/ssonzk/jpa-1%EC%B0%A8-%EC%BA%90%EC%8B%9C%EC%99%80-2%EC%B0%A8-%EC%BA%90%EC%8B%9C-401ac9183a1d

**Comment by JoisFe** [comment]
컴퓨터 구조가 떠오르네요

<img width="573" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/90208100/a8b8e97a-42e7-4a2c-b30b-d2d42108782d">

**Reply by Irisation23** [reply]
<img width="813" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/53285909/f9c9dad5-be53-483d-be8c-3fdf49ed6af3">

**Comment by JoisFe** [comment]
자바 volatile 또한 떠오르네요

<img width="499" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/90208100/7c8f2296-60ce-463c-ba2e-3a7838a84a0f">

[참고자료](https://nesoy.github.io/articles/2018-06/Java-volatile)

- 여기서 몇차 캐시인지는 잘 모르겠으나
